apps/samradsgatt/graphql/withApollo.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { useRouter } from 'next/router'
+import { NextPageContext } from 'next'
+import { ApolloProvider } from '@apollo/client/react'
+import initApollo from './client'
+
+export const withApollo = (Component) => {
+  const NewComponent = ({ apolloState, pageProps }) => {
+    const { asPath } = useRouter()
+    return (
+      <ApolloProvider client={initApollo({ ...apolloState })}>
+        <Component {...pageProps} />
+      </ApolloProvider>
+    )
+  }
+  return NewComponent
+}
+export const getServerSideProps = async (ctx, Component) => {
+  const apolloClient = initApollo({})
+  const newContext = { ...ctx, apolloClient }
+  const props = Component.getInitialProps
+    ? await Component.getInitialProps(newContext)
+    : {}
+  const cache = apolloClient.cache.extract()
+  return {
+    pageProps: props,
+    apolloState: cache,
+    apolloClient,
+  }
+}
```

Comment:
This looks pretty good, but is it used anywhere?

Reply:
I kept the api minimal in apps/samradsgat. Focusing on making one call work.
withApollo is therefore only used in pages/test as of now but will be implemented where needed